core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1853,15 +1856,31 @@ private[spark] class DAGScheduler(
       case FetchFailed(bmAddress, shuffleId, _, mapIndex, reduceId, failureMessage) =>
         val failedStage = stageIdToStage(task.stageId)
         val mapStage = shuffleIdToMapStage(shuffleId)
+        val isExecutorDecommissioned =
```

Comment:
We should not have this overhead here. Could you move this to the `else` statement which needs this variable?

Reply:
Should we use a method like `def isExecutorDecommissioned(bmAddress: BlockManagerId) : Boolean = {……} `  to instead of this variable ？